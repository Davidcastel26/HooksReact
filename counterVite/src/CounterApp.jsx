import React from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

  const funtionForClick = (e) => {
    console.log('hey')
  }

  return (
    <>
      <h1> Counter App </h1>
      <h2> { value } </h2>
      <button onClick={ funtionForClick } > +1 </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}