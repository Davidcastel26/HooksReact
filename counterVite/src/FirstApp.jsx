import React from 'react'

// const newMessage = {
//   message: 'Hola mundo',
//   title: 'Fernando'
// }
// utilizar variables fijate que no cambian afuera del componente para que no se ester recargando esa info

import PropTypes from 'prop-types';

export const FirstApp = ({
  title,
  subTitle,
  name
  // subTitle = 'the end of the word' 
}) => {
  
  // const something = (a , b) => {
  //   return a + b;
  // }
  //Utilizar variables que son cambiantes dentro el componente
  // if (!title){
  //   throw new Error("the title does not exist")
  // }
  
  return (
    <>
      <h1>{title}</h1>
      {/* <h1> { something(1,2) } </h1> */}
      <h2>{ subTitle }</h2>
      {/* <p>Soy un subtitulo</p> */}
      <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  title: 'There is no title',
  subTitle: 'There is no subtitle',
  name: 'David'
}