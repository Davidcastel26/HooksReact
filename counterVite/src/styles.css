
html, body{
    font-size: 1.3rem;
    background-color:  #21232a;
    color: white;
    padding: 70px;
    font-family: Helvetica, Arial, sans-serif;   
}