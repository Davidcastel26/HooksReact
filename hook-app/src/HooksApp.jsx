
export default function HooksApp() {
  return (
    <div>HooksApp</div>
  )
}
