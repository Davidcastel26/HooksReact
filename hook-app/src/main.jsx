import React from 'react'
import ReactDOM from 'react-dom/client'
// import HooksApp from './HooksApp'

import './index.css'
// import CounterApp from './01-useState/CounterApp'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import SimpleForm from './02-useEffect/SimpleForm'
// import { FomrWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { MemorizeComp } from './06-memos/MemorizeComp'
// import { MemoHook } from './06-memos/MemoHook'
// import { CallbackHook } from './06-memos/CallbackHook'
import { Padre } from './07-tarea-memo/Padre'

import './08-useReducer/intro-useReducer';

// ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    // <Padre />
  // </React.StrictMode>,
// )
