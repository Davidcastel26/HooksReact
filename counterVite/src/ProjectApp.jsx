import { FirstApp } from './FirstApp'
// import { CounterApp } from './CounterApp';

export const App = () => {
    return (
        <div>
            {/* <CounterApp value={20}/> */}
            <FirstApp  title="Hola soy yo"/>
        </div>
    );   
}