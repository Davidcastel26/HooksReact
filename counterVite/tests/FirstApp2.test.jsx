import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('testing component <FirstApp/>', () => { 

    const title = "Hola, Soy Goku";
    const subTitle = 'Soy un subTitle'

    test('should be match with the snapsho',() => {
        
        const { container } = render( <FirstApp title={ title } />)
        expect( container ).toMatchSnapshot();

    })

    test('it should show a message Hola, Soy Goku', () => {
        
        render( <FirstApp title={title} />);
        //screen.debug
        expect( screen.getByText(title) ).toBeTruthy()

    })

    test('should show the title into h1', () => { 
        
        render( <FirstApp title={title} />);
        expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain( title )

     })

    test('should show the subtitle sent by props', () => { 
        
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />  
        );

        expect( screen.getAllByText(subTitle).length ).toBe(1);

     })
})
        
    