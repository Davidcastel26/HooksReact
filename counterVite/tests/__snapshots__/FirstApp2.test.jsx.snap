// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <FirstApp/> should be match with the snapsho 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
     
    Theres no subtitle
     
  </p>
  <p>
    David
  </p>
</div>
`;
